Lib/test/test_tarfile.py

```diff
@@ -2498,6 +2498,70 @@ def test_partial_input_bz2(self):
         self._test_partial_input("r:bz2")
 
 
+class SymlinkOverwriteTest(unittest.TestCase):
+    # The testcase checks for correct overwriting of an
+    # existing symlink (issue #12800)
+
+    @support.skip_unless_symlink
+    def test_overwrite_symlink(self):
+        tmpdir = support.temp_cwd('overwrite_symlink')
+        source = 'source'
+        link = 'link'
+        try:
+            with open(source, 'wb'):
+                pass
+            os.symlink(source, link)
+            with tarfile.open(tmpname, 'w') as tar:
+                tar.add(source, arcname=os.path.basename(source))
+                tar.add(link, arcname=os.path.basename(link))
+
+            with open(tmpname, 'rb') as fileobj:
+                with tarfile.open(fileobj=fileobj, mode='r|') as tar:
+                    tar.extractall(path=support.SAVEDCWD)
+        finally:
+            try:
+                os.unlink(link)
+            except:
+                pass
+
+            try:
+                os.unlink(source)
+            except:
+                pass
```

Comment:
If using a temporary directory for your test (temp_cwd in a context manager for example), you should not need this at all.